pandas/core/series.py

```diff
@@ -447,8 +447,12 @@ def __init__(
                     data = SingleArrayManager.from_array(data, index)
 
         generic.NDFrame.__init__(self, data)
-        self.name = name
-        self._set_axis(0, index, fastpath=True)
+        if fastpath:
```

Comment:
`_set_axis` is not being called for `fastpath=True`, so there is no logic to put in there. 

Of course I could pass through fastpath to `_set_axis`, and let it start with `if fastpath: return` so that it becomes a no-op. But I don't really see the benefit of that, since I need to have a `if fastpath: .. else: ..` here anyway to handle the `name`